pkg/lib/serverTime.js

```diff
@@ -368,10 +368,23 @@ export function ServerTime() {
 
         const text = "# This file is automatically generated by Cockpit\n\n" + config.servers.map(s => `server ${s}\n`).join("");
 
-        if (config.enabled)
-            return enabled_file.replace(text).then(() => disabled_file.replace(null));
-        else
-            return disabled_file.replace(text).then(() => enabled_file.replace(null));
+        function ensure_sourcedir() {
+            function add_sourcedir(data) {
```

Comment:
FTR: I'm fine with that approach. Please add the bugzilla ref as a comment once you have it.

Reply:
Done.